streaming-common/src/main/java/com/sequenceiq/cloudbreak/streaming/processor/AbstractRecordProcessor.java

```diff
@@ -0,0 +1,155 @@
+     */
+    @PostConstruct
+    public void init() {
+        if (!isProcessingEnabled()) {
+            LOGGER.debug("Processing is not enabled for {}", getServiceName());
+        } else if (!validateConfiguration(configuration)) {
+            LOGGER.debug("Configuration failed for {} processing.", getServiceName());
+        } else {
+            LOGGER.debug("Starting record worker for {} processing service.", getServiceName());
+            RoundRobinStreamProcessingQueues<C, R, W> processingQueues = new RoundRobinStreamProcessingQueues<>(
+                    numberOfWorkers, queueSizeLimit, this);
+            processingQueuesRef.set(processingQueues);
+            getProcessingQueues().startWorkers();
+        }
+    }
+
+    /**
+     * Creating a new custom worker (client operations needs to be implemented in the worker)
+     * @param threadName thread name of the worker that is calculsted by the round robin processing queue.
+     * @param processingQueue processing queue that the worker watches.
+     * @return newly created worker
+     */
+    public abstract W createWorker(String threadName, BlockingDeque<R> processingQueue);
+
+    /**
+     * Name that represents the processing service type. It should be unique for every processor.
+     */
+    public abstract String getServiceName();
+
+    /**
+     * Override this to change the default behavior when the processing queue is too high.
+     * Default behaviour: log the dropped record request input.
+     */
+    public void handleDroppedRecordRequest(R input, int sizeLimit) {
+        LOGGER.warn("Blocking queue reached size limit: {}. Dropping record input: {}", sizeLimit, input);
+    }
+
+    /**
+     * Override this to change the default behavior when an stream record processing error happens.
+     * Default behaviour: log the exception.
+     */
+    public void handleDataStreamingException(R input, StreamProcessingException e) {
+        LOGGER.warn(String.format("Exception during stream record processing [skip] - input: %s", input), e);
+    }
+
+    /**
+     * Override this to change the default behavior when an unexpected error happens.
+     * Default behaviour: log the exception.
+     */
+    public void handleUnexpectedException(R input, Exception e) {
+        LOGGER.warn(String.format("Unexpected exception during stream record processing [skip] - input: %s", input), e);
+    }
+
+    public boolean isProcessingEnabled() {
+        return processingEnabled.get();
+    }
+
+    public RoundRobinStreamProcessingQueues<C, R, W> getProcessingQueues() {
+        return processingQueuesRef.get();
+    }
+
+    /**
+     * Validate provided custom configuration.
+     * @param configuration object that holds global configs for the processor.
+     * @return validation result
+     */
+    public boolean validateConfiguration(C configuration) {
```

Comment:
Why is the config passed as parameter? Couldn't overriding objects simply validate this.configuration?

Reply:
could be but it's more clear with this way with an abstraction like this to know you need to do something on that configuration input